kiosk/sources/extensions/collection_bidding_ext.move

```diff
@@ -0,0 +1,220 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+/// Implements Collection Bidding. Currently it's a Marketplace-only functionality.
+///
+/// It is important that the bidder chooses the Marketplace, not the buyer.
+module kiosk::collection_bidding_ext {
+    use std::option::Option;
+    use std::type_name;
+    use std::vector;
+
+    use sui::kiosk::{Self, Kiosk, KioskOwnerCap};
+    use sui::kiosk_extension as ext;
+    use sui::tx_context::TxContext;
+    use sui::coin::{Self, Coin};
+    use sui::transfer_policy::{
+        Self as policy,
+        TransferPolicy,
+        TransferRequest,
+    };
+    use sui::sui::SUI;
+    use sui::vec_set;
+    use sui::object::{Self, ID};
+    use sui::event;
+    use sui::pay;
+    use sui::bag;
+
+    use kiosk::personal_kiosk;
+    use kiosk::kiosk_lock_rule::Rule as LockRule;
+    use kiosk::marketplace_adapter::{Self as mkt, MarketPurchaseCap, NoMarket};
+
+    /// Trying to perform an action in another user's Kiosk.
+    const ENotAuthorized: u64 = 0;
+    /// Trying to accept the bid in a disabled extension.
+    const EExtensionDisabled: u64 = 1;
+    /// A `PurchaseCap` was created in a different Kiosk.
+    const EIncorrectKiosk: u64 = 2;
+    /// The bid amount is less than the minimum price.
+    const EIncorrectAmount: u64 = 3;
+    /// Trying to accept a bid using a wrong function.
+    const EIncorrectMarketArg: u64 = 4;
+    /// Trying to accept a bid that does not exist.
+    const EBidNotFound: u64 = 5;
+    /// Trying to place a bid with no coins.
+    const ENoCoinsPassed: u64 = 6;
+
+    /// A key for Extension storage - a single bid on an item of type `T` on a `Market`.
+    struct Bid<phantom T, phantom Market> has copy, store, drop {}
+
+    // === Events ===
+
+    /// An event that is emitted when a new bid is placed.
+    struct NewBid<phantom T, phantom Market> has copy, drop {
+        kiosk_id: ID,
+        bids: vector<u64>,
+        kiosk_owner: Option<address>,
+    }
+
+    /// An event that is emitted when a bid is accepted.
+    struct BidAccepted<phantom T, phantom Market> has copy, drop {
```

Comment:
Should we also include the amount? Might be helpful for indexing

Reply:
Agreed, this could help indexing!